app/src/server/api/external/v1Api.router.ts

```diff
@@ -115,62 +103,89 @@ export const v1ApiRouter = createOpenApiRouter({
           : chatCompletionInputReqPayload.parse(input);
 
       const modelSlug = inputPayload.model.replace("openpipe:", "");
-      const fineTune = await prisma.fineTune.findUnique({
-        where: { slug: modelSlug },
-      });
-
-      if (!fineTune) {
-        throw new TRPCError({ message: "The model does not exist", code: "NOT_FOUND" });
-      }
+      const fineTune =
+        (await prisma.fineTune.findUnique({
+          where: { slug: modelSlug },
+        })) ?? undefined;
 
-      const typedFT = typedFineTune(fineTune);
-
-      if (typedFT.projectId !== key.projectId) {
+      if (inputPayload.model.startsWith("openpipe:") && !fineTune) {
```

Comment:
The flow of this function feels somewhat hard to follow with me. I think it would be cleaner to just have an `if (inputPayload.model.startswith("openpipe:"))` at the top and split at that point, instead of having some checks here and then a separate `if (fineTune)` down below.

Reply:
I refactored it